### ISSUE TITLE
slow_vector_initialization emits an internal warning ("rustc_errors::emitter Invalid span")

### ISSUE THREAD

**Issue by goffrie** [rust-lang/rust-clippy#11408]
### Summary

I see this warning message printed:

```
WARN rustc_errors::emitter Invalid span /rustc/58eefc33adf769a1abe12ad94b3e6811185b4ce5/library/alloc/src/macros.rs:44:36: 44:59 (#4), error=SourceNotAvailable { filename: Real(Remapped { local_path: None, virtual_name: "/rustc/58eefc33adf769a1abe12ad94b3e6811185b4ce5/library/alloc/src/macros.rs" }) }
```

(This is also confusing my tooling that uses JSON output format since this line is not JSON :p)

### Reproducer

I tried [this code](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=c07cd6613fccb2ef29d5cca88adf9197):

```rust
pub fn wat() {
    let mut contents = vec![];
    contents.resize(123, 0);
}
```

In addition to a (correct) lint message, it produces the warning.

### Version

```text
1.74.0-nightly (2023-08-24 58eefc33adf769a1abe1)

(per playground, I didn't test locally)
```

**Comment by y21** [comment]
We probably need to put a `.source_callsite()` on [here](https://github.com/rust-lang/rust-clippy/blob/706c48b62afb036d34dfa3d6119846edb0afda63/clippy_lints/src/slow_vector_initialization.rs#L208) so it gets the span of where `vec![]` was invoked and not some span within the macro expansion.
